public/app/features/alerting/unified/mocks/server/handlers/k8s/timeIntervals.k8s.ts

```diff
@@ -59,6 +59,17 @@ export const listNamespacedTimeIntervalHandler = () =>
           { status: 403 }
         );
       }
+
+      // Rudimentary filter support for `spec.name`
```

Comment:
Even if the back-end doesn't support all of the filters yet we should create a function that replicates the behaviour with the assumption that any key can match; something along these lines should work

```TypeScript
const items = [{ spec: { foo: 'bar', baz: 'qux' } }, { spec: { foo: 'baz' } }];
const selector = 'spec.foo=bar, spec.baz=qux'

// selector = 'spec.foo=bar, spec.baz=qux';
// items = [{ spec: { foo: 'bar', baz: 'qux' } }, { spec: { foo: 'baz' } }];
function filterBySelector<T>(items: T[], selector: string) {
  // [['path.to.key', 'value'], ['other.path', 'value']]
  const filters: Array<string[]> = chain(selector)
    .split(',')
    .map(trim)
    .map(s => s.split('='))
    .value();
  
  return filter(items, item => filters.every(([key, value]) => {
    const matcher = matchesProperty(key, value)
    return matcher(item)
  }))
}

filterBySelector(items, selector)
// returns [ { spec: { foo: 'bar', baz: 'qux' } } ]
```

Reply:
:+1: I'll add this in to generalise it now (thanks), but I'm going to make the handler still verify if the selector contains something we know that the real API will filter by, so we maintain consistency with the real implementation

Reply:
Hm that does sound like a terrible maintenance burden having to synchronize the available filters between both teams though I get the appeal of matching the mock implementation as close as we can to avoid writing tests that pass with the mock but not the actual back-end implementation.

Reply:
I think it's okay because it makes it clear what requirements the UI code is working to - if we write new functionality that relies on a different filter, and we need a test for it, then we should inherently be updating those filters in the MSW implementation :+1:

Reply:
Yea that's definitely a sensible way forward – and with new filters added we should be writing tests for those too in the FE :D